### PR TITLE
Add Rstudio addin for package and reset/highlight graph nodes

### LINKED ISSUE #8: Consider adding RStudio Addin
## Prework

* [x] I understand and agree to `targetsketch`' [code of conduct](https://github.com/wlandau/targetsketch/blob/main/CODE_OF_CONDUCT.md).
* [x] I understand and agree to `targetsketch`' [contributing guidelines](https://github.com/wlandau/targetsketch/blob/main/CONTRIBUTING.md).
* [x] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.

## Proposal

Please describe the new feature. If applicable, write a minimal example in R code or pseudo-code to show input, usage, and desired output. To help us read any code you include (optional) please try to follow the [tidyverse style guide](https://style.tidyverse.org/). The `style_text()` and `style_file()` functions from the [`styler`](https://github.com/r-lib/styler) package make it easier.

Addins are accessible from the RStudio command palette (Ctrl/Cmd + Shift + P). It would prob be useful for this package. Here is the possible registration metadata for the addin. I can do a PR if that's okay.
`inst/rstudio/addins.dcf`
```
Name: targets builder
Description: R/Shiny app companion to targets R package.
Binding: targetsketch
Interactive: true
```

### LINKED ISSUE #10: Highlight graph target on manifest row selection
## Prework

* [x] I understand and agree to `targetsketch`' [code of conduct](https://github.com/wlandau/targetsketch/blob/main/CODE_OF_CONDUCT.md).
* [x] I understand and agree to `targetsketch`' [contributing guidelines](https://github.com/wlandau/targetsketch/blob/main/CONTRIBUTING.md).
* [x] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.

## Proposal

Please describe the new feature. If applicable, write a minimal example in R code or pseudo-code to show input, usage, and desired output. To help us read any code you include (optional) please try to follow the [tidyverse style guide](https://style.tidyverse.org/). The `style_text()` and `style_file()` functions from the [`styler`](https://github.com/r-lib/styler) package make it easier.

Consider changing color for the target in the Graph card when a row is selected within the Manifest card dataTable. This would make it easy to quickly see the target position in the dependency graph. The feature could be toggled on and off via a checkbox, if needed. I can work on this if that's okay.

### LINKED ISSUE #11: Add button to reset graph card view to default view
## Prework

* [x] I understand and agree to `targetsketch`' [code of conduct](https://github.com/wlandau/targetsketch/blob/main/CODE_OF_CONDUCT.md).
* [x] I understand and agree to `targetsketch`' [contributing guidelines](https://github.com/wlandau/targetsketch/blob/main/CONTRIBUTING.md).
* [x] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.

## Proposal

Please describe the new feature. If applicable, write a minimal example in R code or pseudo-code to show input, usage, and desired output. To help us read any code you include (optional) please try to follow the [tidyverse style guide](https://style.tidyverse.org/). The `style_text()` and `style_file()` functions from the [`styler`](https://github.com/r-lib/styler) package make it easier.

Enhancement for user who may have inadvertently zoomed in or moved the dependency graph. Reset Graph View button in Control card or Graph card would update visnetwork back to version prior to zoom or mouse drag actions. I can work on this if that's okay.

### PR DESCRIPTION
**Pull request by parmsam**
# Prework

* [x] I understand and agree to this repository's [code of conduct](https://github.com/wlandau/targetsketch/blob/main/CODE_OF_CONDUCT.md).
* [x] I understand and agree to this repository's [contributing guidelines](https://github.com/wlandau/targetsketch/blob/main/CONTRIBUTING.md).
* [x] I have already submitted an issue to the [issue tracker](http://github.com/wlandau/targetsketch/issues) to discuss my idea with the maintainer.

# Summary

Please explain the purpose and scope of your contribution.

* Adds in RStudio Addin for targetsketch::targetsketch() using dcf file.
* Updated News.md
* Fixed #8, 
* Fixed #10 
* Fixed #11 

# Related GitHub issues and pull requests

* Ref: #8, #10, #11 

# Checklist

* [x] This pull request is not a [draft](https://github.blog/2019-02-14-introducing-draft-pull-requests).

### CODE REVIEW

File: inst/rstudio/addins.dcf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Name: targets builder
+Description: R/Shiny app companion to targets R package.
+Binding: targetsketch
+Interactive: true
```
Comment (wlandau) [inline inst/rstudio/addins.dcf:4]: Would you add a newline at the end of the file?

File: R/server.R [hunk 2]
```diff
@@ -5,6 +5,29 @@ server <- function(input, output, session) {
...
+        visNetwork::visUnselectAll() |>
+        visNetwork::visUpdateNodes(nodes= data.frame(id = prevSelectedIDs, color = "#899DA4"))
+    }
+  )
+  shiny::observeEvent(
+    input$manifest_rows_selected, {
+      print( values$manifest[input$manifest_rows_selected, ] )
+      allIDs <- values$manifest$name
+      selectedIDs <- values$manifest[input$manifest_rows_selected, ]$name
+      visNetwork::visNetworkProxy("graph") |>
```
Comment (wlandau) [inline R/server.R:24]: The native pipe requires R >= 4.1.0. Would you update the `DESCRIPTION` file to reflect this?

File: R/ui.R [hunk 3]
```diff
@@ -21,8 +21,12 @@ ui <- function() {
               "Update",
               icon = shiny::icon("redo-alt")
             ),
-            shiny::downloadButton("download", "Download")
-          ),
+            shiny::downloadButton("download", "Download"),
+            shiny::actionButton(
+              "reset", 
+              "Reset",
```
Comment (wlandau) [inline R/ui.R:27]: With the growing number of buttons, we may need to be more specific in the titles: for example, "Reset selection".

File: R/server.R [hunk 4]
```diff
@@ -5,6 +5,29 @@ server <- function(input, output, session) {
     update_values(values, input),
     ignoreNULL = FALSE
   )
+  shiny::observeEvent(
+    input$reset,{
```
Comment (wlandau) [inline R/server.R:9]: If I click "Reset" with nothing selected, I get an error:

```r
# A tibble: 0 × 2
# … with 2 variables: name <chr>, command <chr>
# ℹ Use `colnames()` to see all variable names
# A tibble: 0 × 2
# … with 2 variables: name <chr>, command <chr>
# ℹ Use `colnames()` to see all variable names
Warning: Error in data.frame: arguments imply differing number of rows: 0, 1
[No stack trace available]
```

**Review by wlandau — CHANGES_REQUESTED** [review]
Looks good, my inline comments are minor.

File: R/ui.R (hunk 3 above)
Comment (parmsam) [reply R/ui.R]: Changed button label to Reset selection.

File: R/server.R (hunk 2 above)
Comment (parmsam) [reply R/server.R:26]: Done.

File: inst/rstudio/addins.dcf (hunk 1 above)
Comment (parmsam) [reply inst/rstudio/addins.dcf]: Added.

File: R/server.R (hunk 4 above)
Comment (parmsam) [reply R/server.R:9]: It should allow you to click reset with nothing selected now.

**Review by wlandau — APPROVED** [review]

**Comment by wlandau** [comment]
This PR also has lints: https://github.com/wlandau/targetsketch/runs/8250736793?check_suite_focus=true#step:7:15

**Comment by wlandau** [comment]
By the way, you can lint locally with `lintr::lint_package()`. The `targetsketch` `.lintr` file might not be up to date with the new version of `lintr`. `targets` uses this one: https://github.com/ropensci/targets/blob/main/.lintr

**Comment by wlandau** [comment]
Looks like there are merge conflicts with the other PR. Would you resolve them?

**Pull request merged**